### ISSUE TITLE
[RLlib] PPO surrogate loss incorrectly clipped

### ISSUE THREAD

**Issue by MattiasDC** [ray-project/ray#37334]
### What happened + What you expected to happen

In both the torch and TF implementations the surrogate loss calculation is incorrectly clipped. The `clip_param` should normally clip the `logp_ratio` (by default between 0.7 and 1.3). However when you look at the torch implementation:

```
surrogate_loss = torch.min(
            train_batch[Postprocessing.ADVANTAGES] * logp_ratio,
            train_batch[Postprocessing.ADVANTAGES]
            * torch.clamp(
                logp_ratio, 1 - self.config["clip_param"], 1 + self.config["clip_param"]
            ),
        )
```

You can see that the `torch.min` is taken. In case of negative advantages, a big positive `logp_ratio` in combination with a negative advantage, this will result an a huge negative `surrogate_loss`. In my environment this could amount to even billions of negative surrogate loss.

A proposed fix for torch, but this should also be fixed for tensorflow.

```
absolute_advantages = torch.abs(train_batch[Postprocessing.ADVANTAGES])
        clamped_logp_ratio = torch.clamp(
            logp_ratio, 1 - self.config["clip_param"], 1 + self.config["clip_param"])
        surrogate_loss = torch.sign(train_batch[Postprocessing.ADVANTAGES]) * torch.min(
            absolute_advantages * logp_ratio,
            absolute_advantages * clamped_logp_ratio)
```

### Versions / Dependencies

RlLib 3.0.0

### Reproduction script

Not needed

### Issue Severity

High: It blocks me from completing my task.

**Comment by ArturNiederfahrenhorst** [comment]
This needs more discussion wether our loss is actually incorrect, see https://github.com/ray-project/ray/pull/37454/

**Comment by MattiasDC** [comment]
Closing, as per #37454
